src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2180,7 +2180,7 @@ private BoundExpression BindUnaryOperatorCore(CSharpSyntaxNode node, string oper
         {
             UnaryOperatorKind kind = SyntaxKindToUnaryOperatorKind(node.Kind());
 
-            bool isOperandTypeNull = (object)operand.Type == null;
+            bool isOperandTypeNull = operand.IsLiteralNull();
             if (isOperandTypeNull)
```

Comment:
Are we reporting an error for an operand with no type, for instance `BoundMethodGroup`?